packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts

```diff
@@ -205,16 +219,21 @@ export class CryptoWatchPriceFeed extends PriceFeedInterface {
     return this.priceFeedDecimals;
   }
 
-  public async update(): Promise<void> {
+  public async update(ancillaryData?: string): Promise<void> {
     const currentTime = await this.getTime();
 
-    // Return early if the last call was too recent.
-    if (this.lastUpdateTime !== null && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+    // Return early if the last call for this ancillary data was too recent.
+    if (
+      this.lastUpdateTime !== null &&
+      this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime &&
+      ancillaryData === this.lastAncillaryData
```

Comment:
Now that `ancillaryData` can be passed into `update`, we should only exit prematurely if the last update was for a recent timestamp AND the ancillary data is the same as the last one queried